### ISSUE TITLE
Unable to install webpack@latest because of updated @long dependency

### ISSUE THREAD

**Issue by Ivaylo-Lafchiev** [xtuc/webassemblyjs#486]
> npm sill fetchPackageMetaData error for long@git://github.com/dcodeIO/long.js.git#8181a6b50a2a230f0b2a1e4c4093f9b9d19c8b69 Error while executing: 

In my firm direct access to Github assets is blocked, and I imagine this will be the case for a lot of enterprise applications. Hence it is impossible to install the latest version of Webpack which depends on 1.8.2 of this module. 

Can this change be reverted or fixed?

**Comment by xtuc** [comment]
I didn't though about that, sorry! I'll change it.

**Comment by xtuc** [comment]
cc @dcodeIO would you mind doing a release of long.js?

**Comment by JacobGrady** [comment]
Same issue (I think) preventing me from installing Webpack ina  new project

```
94 silly resolveWithNewModule @webassemblyjs/wast-parser@1.8.2 checking installable status
95 silly fetchPackageMetaData error for long@git://github.com/dcodeIO/long.js.git#8181a6b50a2a230f0b2a1e4c4093f9b9d19c8b69 Error while executing:
95 silly fetchPackageMetaData undefined ls-remote -h -t git://github.com/dcodeIO/long.js.git
95 silly fetchPackageMetaData
95 silly fetchPackageMetaData
95 silly fetchPackageMetaData spawn git ENOENT
96 verbose stack RangeError: Maximum call stack size exceeded
96 verbose stack     at RegExp.test (<anonymous>)
96 verbose stack     at isDepOptional (C:\Users\Jacob\AppData\Roaming\npm\node_modules\npm\lib\install\deps.js:405:45)
96 verbose stack     at failedDependency (C:\Users\Jacob\AppData\Roaming\npm\node_modules\npm\lib\install\deps.js:414:9)
```

**Comment by xtuc** [comment]
@JacobGrady do you have `git` installed on your system? If not could you try installing it?

**Comment by JacobGrady** [comment]
@xtuc I did not have git installed, did install after you mentioned it, and the webpack install is now working properly. Thank you!

However I believe webpack had been previously installed via npm without git being installed on my system without any problems (at least a year ago)

**Comment by amadt** [comment]
Seeing the same error all day today. I do have git installed though:
```
> which git
/usr/local/bin/git
```
Any ideas?

**Comment by Quinncuatro** [comment]
I'm getting the same error, @amadt . @xtuc - I have git installed and it's not going through.

**Comment by amadt** [comment]
I'm on a corporate network. Once I tried from outside my network it succeeded.

**Comment by markula** [comment]
The issue I ran into is my app installs and builds modules inside a docker image where it makes no sense to have a git installed or user credentials exposed. This has killed deploys today with no obvious work around.

**Comment by dcompiled** [comment]
Trying to install webpack for the first time as a new user and blocked with this error: "Maximum call stack size exceeded"

I have git installed, it simply does not work.  Testing this from behind a corporate network.  The command that fails specifically is:

107 silly fetchPackageMetaData C:\Users\......\Programs\Git\cmd\git.EXE ls-remote -h -t git://github.com/dcodeIO/long.js.git
107 silly fetchPackageMetaData
107 silly fetchPackageMetaData fatal: read error: Invalid argument

I can run the command with https:// but not git:// just fine.  

git.EXE ls-remote -h -t git://github.com/dcodeIO/long.js.git

I am a bit confused why git is required to operate on the command line to download this npm package.  Is there any way I can force it to use https?

**Comment by xtuc** [comment]
As a workaround you can install `webpack@4.29.3`

**Comment by sagitsofan** [comment]
Same error here:

4450 silly fetchPackageMetaData C:\Program Files\Git\cmd\git.EXE ls-remote -h -t git://github.com/dcodeIO/long.js.git
4450 silly fetchPackageMetaData
4450 silly fetchPackageMetaData fatal: unable to connect to github.com:
4450 silly fetchPackageMetaData github.com[0: 192.30.253.113]: errno=No error
4450 silly fetchPackageMetaData github.com[1: 192.30.253.112]: errno=No error
4450 silly fetchPackageMetaData
4450 silly fetchPackageMetaData
4450 silly fetchPackageMetaData exited with error code: 128
4451 verbose stack RangeError: Maximum call stack size exceeded
4451 verbose stack     at RegExp.test (<anonymous>)

**Comment by shayantabatabaee** [comment]
I have same issue too, is there any solution for this ?

**Comment by sagitsofan** [comment]
Install `webpack` with specific version `4.29.3`

**Comment by evbermudez** [comment]
> As a workaround you can install `webpack@4.29.3`

This works for me. Thank you with <3

**Comment by geocine** [comment]
Can you not use the ssh protocol, just use the https procotol for cloning? I build most build systems have ssh ports disabled.

**Comment by joeljeske** [comment]
Hooray thanks!

**Comment by markula** [comment]
Thanks for patching this. Much appreciated.

**Comment by darekkay** [comment]
Thanks! `webpack` [released](https://www.npmjs.com/package/webpack/v/4.29.5) a patched version `4.29.5` today. Don't forget to **expire your registry cache** if you are using a custom (corporate) registry.

**Comment by bc-lee** [comment]
Because of this issue, I was converting my project to use yarn.
```
//package.json
"resolutions": {
    "**/@webassemblyjs/**/long": "4.0.0"
},
```

BTW, https://www.npmjs.com/package/@xtuc/long/v/4.2.2 says there is a version named `4.2.2`, but I cannot find any codes or tags for such version in https://github.com/xtuc/long.js. May I ask why there are no tags for this release?
